platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/location/LocationComponent.java

```diff
@@ -890,11 +908,14 @@ private void onLocationLayerStop() {
     mapboxMap.removeOnCameraIdleListener(onCameraIdleListener);
   }
 
-  private void initialize(@NonNull Context context, @NonNull LocationComponentOptions options) {
-    if (isInitialized) {
+  private void initialize(@NonNull final Context context, @NonNull Style style,
+                          @NonNull final LocationComponentOptions options) {
+    if (isComponentInitialized) {
```

Comment:
Can we return and possibly throw an exception if somebody is passing an invalid style (`styleLoaded == false`)?